pkg/scheduler/framework/plugins/nodeaffinity/node_affinity.go

```diff
@@ -83,10 +85,43 @@ func (s *preFilterState) Clone() framework.StateData {
 // failed by this plugin schedulable.
 func (pl *NodeAffinity) EventsToRegister() []framework.ClusterEventWithHint {
```

Comment:
Aside as I missed the original PR about ClusterEventWithHint. QueueingHintFn will be awkward, if possible, to define in webassembly, as dynamic functions don't work the same way from compiler to compiler.

It is further complicated by the signature accepting interface{}. I think basically this will be out of scope, where a different signature more natural like the others would be easier.

For example, a normal extension with a plugin-defined "queueHintID" which is a field of ClusterEvent, and a parameter to that function. Particularly, that function should not be open types like interface{}

Reply:
I don't yet understand from your explanation, how we can eliminate `interface{}` from the parameters of QueueingHintFn. Could you elaborate with some examples?